### PR TITLE
Pagination with numbers

### PR DESCRIPTION
**Pull request by axton21**
@DFurnes @angaither 
Pagination with numbers. Looks much better than before and move it to the bottom of the table.

### CODE REVIEW

**Comment by axton21** [comment]
screen shot 2015-07-25 at 01 47 58
screen shot 2015-07-25 at 01 48 45
screen shot 2015-07-25 at 01 48 24

File: app/views/users/partials/waypoints.blade.php [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
+    <?php $prev = $data['current_page'] - 1 ?>
+    <li> {{ link_to_route('users.index', 'First', array('page=1')) }} </li>
+    <li> {{ link_to_route('users.index', 'Previous', array('page=' . $prev)) }} </li>
+  @endif
 
- @if ($data['current_page'] > 1)
-   <?php $prev =  $data['current_page'] - 1 ?>
-   <li> {{ link_to_route('users.index', 'Prev', array('page=' . $prev))}} </li>
- @endif
+  <?php $firstLeft = ($data['current_page'] - 5 >= 1) ? $data['current_page'] - 5 : 1 ?>
```
Comment (angaither) [inline app/views/users/partials/waypoints.blade.php]: snake case these vars `first_page`

File: app/views/users/partials/waypoints.blade.php [hunk 2]
```diff
@@ -1,15 +1,26 @@
...
+  <?php $lastRight = ($data['current_page'] + 4 <= $data['last_page']) ? $data['current_page'] + 4 : $data['last_page'] ?>
+  <?php $lastRight = $lastRight < 10 && $data['last_page'] > 10 ? 10 : $lastRight ?>
+  <?php $firstLeft = $firstLeft > 10 && $data['last_page'] > 10 ? $lastRight - 9 : $firstLeft ?>
```
Comment (angaither) [inline app/views/users/partials/waypoints.blade.php]: also, can you add comments about what this logic is doing

**Comment by axton21** [comment]
@angaither @DFurnes 
so overall with this pagination,  the index page will always has ten numbers.

**Comment by angaither** [comment]
:+1:

**Comment by DFurnes** [comment]
For a future refactoring, this would be a great use case for Laravel's [built-in paginator](http://laravel.com/docs/4.2/pagination).

**Comment by DFurnes** [comment]
(:+1: to merge though! Looks great!)

**Comment by axton21** [comment]
@DFurnes does that works for calling API though?

**Comment by DFurnes** [comment]
I think so! You should be able to make a custom Paginator instance (to use Paginator niceness with the API response) & a [custom presenter](http://laravel.com/docs/4.2/pagination#custom-presenters) to format it using Neue's markup. Definitely not required for this to merge though – just a suggestion for future refactoring. :smile:

**Comment by axton21** [comment]
@DFurnes Sure do, thanks!

**Pull request merged**